### ISSUE TITLE
Improve <CompactSearch /> keyboard navigation

### ISSUE THREAD

**Issue by charrondev** [vanilla/vanilla#8096]
- For better accessibility this miniature "form" should have a visually hidden submit button that becomes visible by tabbing to it.
- Tabbing should not redirect the page.
- Tabbing should select the clear element.
